wp-resources/plugins/sil-dictionary-webonary/webonary/Webonary_Cloud.php

```diff
@@ -467,15 +464,29 @@ public static function searchEntries($posts, WP_Query $query): ?array
 		// get the selected parts of speech list
 		$taxonomies = Webonary_Parts_Of_Speech::GetPartsOfSpeechSelected();
 
-		if ($searchText === '') {
-			if (!empty($taxonomies[0])) {
-				// This is a listing by semantic domains
-				$apiParams = array(
-					'text' => $taxonomies[0],
-					'searchSemDoms' => '1'
-				);
-			}
-		} else {
+		if ($searchText === '' && !empty($taxonomies[0])) {
+
+			// This is a listing by semantic domains
+			$apiParams = array(
+				'text' => $taxonomies[0],
+				'searchSemDoms' => '1'
+			);
+
+		}
+		elseif ($searchText === '' && !empty($semantic_domains) && !empty(get_page_by_path('/browse/categories'))) {
+
+			// redirect to the semantic domains page
+			$url_lang = get_query_var('lang');
+			$sem_domain_number = $semantic_domains[0];
+			$sem_domain = Webonary_SemanticDomains::GetDomainName($sem_domain_number, self::getCurrentLanguage());
+			$url = get_site_url() . '/browse/categories/?semdomain=' . urlencode($sem_domain) . '&semnumber=' . $sem_domain_number . '.&lang=en';
+			if ($url_lang != '')
+				$url .= '&lang=' . $url_lang;
```

Comment:
There's already a lang=en in $url. Did you want to replace it or have two lang's?

Reply:
The `get_site_url()` function returns just the base, not the full current URL.